iceoryx_hoofs/include/iceoryx_hoofs/memory/relative_pointer.inl

```diff
@@ -27,25 +27,24 @@ namespace memory
 template <typename T>
 // NOLINTJUSTIFICATION NewType size is comparable to an integer, hence pass by value is preferred
 // NOLINTNEXTLINE(performance-unnecessary-value-param)
-inline RelativePointer<T>::RelativePointer(ptr_t ptr, segment_id_t id) noexcept
-    : m_id(id)
-    , m_offset(computeOffset(ptr))
+inline RelativePointer<T>::RelativePointer(ptr_t const ptr, const segment_id_t id) noexcept
+    : RelativePointer(getOffset(id, ptr), id)
 {
 }
 
 template <typename T>
+// AXIVION Next Construct AutosarC++19_03-A12.1.5 : This is the main c'tor which the other c'tors use
 // NOLINTJUSTIFICATION NewType size is comparable to an integer, hence pass by value is preferred
 // NOLINTNEXTLINE(performance-unnecessary-value-param)
-inline RelativePointer<T>::RelativePointer(offset_t offset, segment_id_t id) noexcept
+inline RelativePointer<T>::RelativePointer(const offset_t offset, const segment_id_t id) noexcept
     : m_id(id)
     , m_offset(offset)
 {
 }
 
 template <typename T>
-inline RelativePointer<T>::RelativePointer(ptr_t ptr) noexcept
-    : m_id(searchId(ptr))
-    , m_offset(computeOffset(ptr))
+inline RelativePointer<T>::RelativePointer(ptr_t const ptr) noexcept
+    : RelativePointer(getOffset(segment_id_t{searchId(ptr)}, ptr), segment_id_t{searchId(ptr)})
```

Comment:
AFAIU this would not be compliant with `A12-1-5` because it is explicitly required to use a delegating c'tor. I think it's not super nice, but an good middle ground.

Reply:
In this case, please create a private constructor like `RelativPointer(prt_t const ptr, const segment_id_t id)` and call that from this ctor. The new ctor would then call the one expecting a `offset_t` and and `segment_id_t`.

Regarding the delegating constructors, I thought we refrained from this but I might not recall correctly.

Reply:
... oh and I think the immediately invoked lambda would still work
````cpp
template <typename T>
inline RelativePointer<T>::RelativePointer(ptr_t ptr) noexcept
    : RelativePointer<T>([ptr] {
        auto id = segment_id_t{searchId(ptr)};
        auto offset = getOffset(id, ptr);
        return RelativePointer{offset, id};
    }())
{
}
```

The copy ctor would be used which is defaulted and should be cheap or even be optimized away :)

Reply:
@elBoberido Nice idea, that worked with a few adaptions for be AUTOSAR compliant ;) Thanks!